src/couch_epi_codechange_monitor.erl

```diff
@@ -0,0 +1,63 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+% http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_epi_codechange_monitor).
+
+-behaviour(gen_server).
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+
+-export([start_link/1]).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Exports
+%% ------------------------------------------------------------------
+
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2, code_change/3]).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Handler) ->
+    gen_server:start_link(?MODULE, [Handler], []).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+
+init([Handler]) ->
+    couch_epi_module_keeper:reload(Handler),
+    {ok, Handler}.
+
+handle_call(_Request, _From, State) ->
+    {reply, ok, State}.
+
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info(_Info, State) ->
+    {noreply, State}.
+
+terminate(_Reason, _State) ->
+    ok.
+
+code_change(_OldVsn, Keeper, _Extra) ->
+    couch_epi_module_keeper:reload(Keeper),
```

Comment:
When we do `couch_epi:register_service(chttpd_epi)` from the supervisor of the application. Among other children we get things like:
```erlang
{"couch_epi_codechange_monitor|my_service|providers", 
    {couch_epi_codechange_monitor, start_link, [couch_epi_functions_gen_my_service]}, 
    permanent, 5000, worker, [couch_epi_functions_gen_my_service, provider1, provider2]}
```
`[couch_epi_functions_gen_my_service, provider1, provider2]` - is a list of modules to monitor. If anyone of them change release_handler would call `couch_epi_codechange_monitor:code_change/3`. We call `couch_epi_module_keeper:reload(Keeper)` in order to regenerate dispatch module if needed.
It is also used for notification via `couch_epi_plugin Module:notify/3`.